src/ipc/ipc4/handler.c

```diff
@@ -311,6 +311,7 @@ int ipc4_pipeline_prepare(struct ipc_comp_dev *ppl_icd, uint32_t cmd)
 	case SOF_IPC4_PIPELINE_STATE_EOS:
 		if (status != COMP_STATE_ACTIVE)
 			return IPC4_INVALID_REQUEST;
+		/* fallthrough */
```

Comment:
`COMPILER_FALLTHROUGH`

Reply:
@lyakh compiler_fallthrough doesn't seem to help with the host build with ipc4. What am I missing?

Reply:
Ok, looking at the definition: https://github.com/thesofproject/sof/blob/992dd676aebc69a97defb7d5361e853919b57d14/xtos/include/sof/compiler_attributes.h#L45-L54 - the only possibility I see for that not to work for you is if in your case the `__attribute__((fallthrough))` definition is used, but that attribute isn't known to your compiler and it's ignoring an unknown attribute. Can you try `-Wignored-attributes` (note, that when building with sparse enabled, the `-Wno-attributes` option is used, which suppresses such warnings)

Reply:
Zephyr has a universal "fallthrough" that we could copy/re-use https://github.com/thesofproject/sof/pull/8383#discussion_r1373810500

cc: @singalsu